File: Android/app/src/main/java/com/bdreiss/trackmyattack/listeners/CauseOnClickListener.java
package com.bdreiss.trackmyattack.listeners;

import android.content.Context;
import android.content.DialogInterface;
import android.provider.ContactsContract;
import android.view.View;

import main.java.com.bdreiss.dataAPI.DataModel;
import main.java.com.bdreiss.dataAPI.EntryNotFoundException;
import main.java.com.bdreiss.dataAPI.Intensity;
import main.java.com.bdreiss.dataAPI.IteratorWithIntensity;

public class CauseOnClickListener extends CustomListener {



    public CauseOnClickListener(Context context, DataModel data){
        super(context, data);
    }
    @Override
    public void onClick(View view) {
        super.onClick(view);

        try {
            if (data.getCauseData(text) instanceof IteratorWithIntensity){
                chooseIntensity(context, (dialogInterface, i) -> data.addCause(text, Intensity.values()[i]));
            }
            else{
                data.addCause(text);
            }
        } catch (EntryNotFoundException e) {
            e.printStackTrace();
        }

        data.save();

    }

    @Override
    public CustomListener copy() {
        CauseOnClickListener newListener = new CauseOnClickListener(context, data);
        newListener.setTextValue(text);
        return newListener;
    }

}

File: Android/app/src/main/java/com/bdreiss/trackmyattack/listeners/CauseOnClickListener.kt
package com.bdreiss.trackmyattack.listeners

import android.content.Context
import android.view.View
import main.java.com.bdreiss.dataAPI.DataModel
import main.java.com.bdreiss.dataAPI.IteratorWithIntensity

//Listener for adding data to causes
class CausesOnClickListener(context: Context, data: DataModel) : CustomListener(context, data) {

    override fun onClick(v: View?) {

        if (data.getCauseData(text) is IteratorWithIntensity)
            chooseIntensity(context){
                    chosenEnumValue ->
                data.addCause(text, chosenEnumValue)
                data.save()
            }
        else
            data.addCause(text)
        data.save()

    }

    override fun clone():CustomListener{
        return CausesOnClickListener(context, data)
    }

}